extension/opampextension/opamp_agent.go

```diff
@@ -320,3 +326,31 @@ func (o *opampAgent) onMessage(_ context.Context, msg *types.MessageData) {
 		o.customCapabilityRegistry.ProcessMessage(msg.CustomMessage)
 	}
 }
+
+func getOSDescription(logger *zap.Logger) string {
+	switch runtime.GOOS {
+	case "darwin":
+		output, err := exec.Command("sw_vers").Output()
+		if err != nil {
+			logger.Error("get darwin OS details using 'sw_vers'", zap.Error(err))
+			return ""
+		}
+		return util.ParseDarwinDescription(string(output))
+	case "linux":
+		output, err := exec.Command("lsb_release", "-d").Output()
+		if err != nil {
+			logger.Error("get linux OS details using 'lsb_release -d'", zap.Error(err))
+			return ""
+		}
+		return util.ParseLinuxDescription(string(output))
+	case "windows":
+		output, err := exec.Command("cmd", "/c", "ver").Output()
+		if err != nil {
+			logger.Error("get windows OS details using 'cmd /c ver'", zap.Error(err))
+			return ""
+		}
+		return util.ParseWindowsDescription(string(output))
+	}
+	logger.Error("unrecognized OS to parse details from")
+	return ""
+}
```

Comment:
Thought on just using a library like gopsutil for this?

It seems reasonable to fallback to just using runtime.GOOS if there's a failure here, also.